test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs

```diff
@@ -597,7 +597,7 @@ public int SampleMethod()
             }
             else
             {
-                Assert.NotEmpty(result.Documents);
+                Assert.Empty(result.Documents);
```

Comment:
mmm is this correct?

Reply:
The compiled assembly in this test doesn't have a reference to a source file an thus the result is always empty. Before my change in the ```Instrumenter``` this test returned a result with a empty document source which is exactly what would create the bug in the report generation.

![image](https://user-images.githubusercontent.com/19378678/106999512-9fd8db80-6786-11eb-81c3-a182b6cd830f.png)

After my change such a ```InstrumenterResult``` isn't created anymore and  that's why now ```result.Documents``` is also empty here.

To make this test the way it was intended the dynamically created assembly must contain sequence points with a reference to the source file. But what ever I tried here the assemblies sequence points never had source file information.

https://github.com/coverlet-coverage/coverlet/blob/4bbcd0bde26939da588544966091485dfa48a038/test/coverlet.core.tests/Instrumentation/InstrumenterTests.cs#L556

The assemblies sequence points were always without source information.

![image](https://user-images.githubusercontent.com/19378678/107000403-1cb88500-6788-11eb-880a-e82c84008ba7.png)

Reply:
got it...pls can you add a comment on top of that assert with this explanation for future reference?